### ISSUE TITLE
Improve push handling 

### ISSUE THREAD

**Issue by ganfra** [element-hq/element-x-android#2910]
Currently, when we receive a push we have to call a method on the notification service : 
```kotlin
notificationService.getNotification(
            userId = sessionId,
            roomId = roomId,
            eventId = eventId
)
```

This is ok when we receive a single push, but when we start receiving several push in a raw (by exemple when you come back from airplane mode), this is very slow and make the catchup really incremental (https://github.com/element-hq/element-x-android/issues/2133). We have a throttle mechanism in the `NotificationDrawerManager` which can mitigate a bit the number of updates in the notification panel, but this is far from ideal.

What could be done, is to ask for several notifications in one request (maybe grouped by roomId?). For example we could batch for 500ms or 1s all the push received and try to resolve all of them. 

Currently we just launch the processing in parallel for each push we receive, so we don't even know in which order it'll finish. 
With batching we could ensure the order is correct by using some metadata we got from firebase : 

```kotlin
RemoteMessage {
/**
   * Gets the time in milliseconds from the Epoch that the message was sent.
   *
   * @return The time that the message was sent
   */
  public long getSentTime()
}
```

**Comment by giomfo** [comment]
@ganfra do you confirm this improvement can be applied to EX-iOS in parallel too?

Do you plan to sum up the received pushes, or to display them like they are currently?

**Comment by ganfra** [comment]
Yes this could be applied to EXI, as a new sdk API will provided it.

The rendering of the notifications won't be changed.

**Comment by jmartinesp** [comment]
The SDK and app changes needed for this have already been merged, so we can mark this as fixed.
